subworkflows/local/03_assemble/assemble_organelles.nf

```
@@ -3,30 +3,35 @@ include { MITOHIFI_MITOHIFI          } from "../../../modules/nf-core/mitohifi/m
 
 workflow ASSEMBLE_ORGANELLES {
     take:
-    ch_raw_assemblies
+    ch_assembly_data // Channel: tuple(meta, assembly / reads )
+    ch_assembly_mode // String: enum('c','r')
 
     main:
     ch_versions = Channel.empty()
-    // Find mitochondria
+    // Attempt MitoHiFi workflow
     // TODO: Need to check options to mitohifi modules.
-    MITOHIFI_FINDMITOREFERENCE( ch_raw_assemblies.map { meta, _assemblies -> [ meta, meta.sample.name ] }.unique() )
-    mitohifi_ch = ch_raw_assemblies
+    MITOHIFI_FINDMITOREFERENCE( ch_assembly_data.map { meta, _assembly_data -> [ meta, meta.sample.name ] }.unique() )
+
+    // Prepare mitohifi_ch based on input data type (mode c: contigs, mode r: reads)
+    mitohifi_ch = ch_assembly_data
         .combine(
             MITOHIFI_FINDMITOREFERENCE.out.fasta
                 .join(MITOHIFI_FINDMITOREFERENCE.out.gb),
             by: 0
         )
-        .multiMap { meta, assembly, mitofa, mitogb ->
-            input: [ meta, assembly.pri_fasta ]
+        .multiMap { meta, assembly_data, mitofa, mitogb ->
+            input: [ meta, ch_assembly_mode == "c" ? assembly_data.pri_fasta : assembly_data ] // If c, use contigs. Else r, use reads.
```

`assembly_data` is a map. It should throw an error if a `path` is expected.